src/pyggdrasil/visualize/_mcmc.py

```diff
@@ -239,3 +239,59 @@ def save_rhat_iteration_AD_DL(
     output_dir.mkdir(parents=True, exist_ok=True)
     # save the figure
     fig.savefig(out_fp, format="svg")  # type: ignore
+
+
+def save_ess_iteration_AD_DL(
+    iteration: list[int],
+    ess_bulk_AD: list[float],
+    ess_bulk_DL: list[float],
+    ess_tail_AD: list[float],
+    ess_tail_DL: list[float],
+    out_fp: Path,
+) -> None:
+    """Save plot of ess vs iteration number to disk.
+
+    Args:
+        iteration: list[int]
+            Iteration numbers.
+        out_fp: Path
+            Output file path.
+
+        TODO: add description
+    """
+
+    # make matplotlib figure, given the axes
+
+    fig, ax = plt.subplots()
+    ax.set_xlabel("Iteration")  # type: ignore
+    # get name of distance measure
+    ax.set_ylabel(r"$ESS$")  # type: ignore
+    # bulk
+    ax.plot(iteration, ess_bulk_AD, color="darkgreen", linestyle="-")  # type: ignore
+    ax.plot(iteration, ess_bulk_DL, color="darkorange", linestyle="-")  # type: ignore
+    # tail
+    ax.plot(iteration, ess_tail_AD, color="darkgreen", linestyle="--")  # type: ignore
+    ax.plot(iteration, ess_tail_DL, color="darkorange", linestyle="--")  # type: ignore
+
+    # artificial legend
+    # add solid line black line to ledgend as bulk
+    ax.plot([], [], color="black", label="bulk", linestyle="-")  # type: ignore
+    # add dashed line black line to ledgend as tail
+    ax.plot([], [], color="black", label="tail", linestyle="--")  # type: ignore
+    # add darkgreen marker to ledgend as AD
+    ax.plot([], [], color="darkgreen", label="AD", marker="o", linestyle="")  # type: ignore
+    # add darkorange marker to ledgend as DL
+    ax.plot([], [], color="darkorange", label="DL", marker="o", linestyle="")  # type: ignore
+
+    # specifying horizontal line type
+    # see limits https://arxiv.org/pdf/1903.08008.pdf
+    # 400 at least
+    plt.axhline(y=400, color="r", linestyle="-", linewidth=0.5)  # type: ignore
```

Comment:
In the future I'd suggest using [ax.axhline](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.axhline.html), rather than `plt.axhline`, but it's not really a big deal in this place.